workspace/src/path_planning/path_planning/path_planning.py

```diff
@@ -207,25 +225,90 @@ def plan_path(self):
 
         return target_pt
 
+    def wpts_path_plan(self):
+        x_current = self.state.pose.position.x
+        y_current = self.state.pose.position.y
+        #theta_current = np.arctan2( 2*(x*w+z*y), x**2-w**2+y**2-z**2 )
+        theta_current = self.state.pose.orientation.z-0.24845347641462115
+        while theta_current<-np.pi:
+            theta_current = theta_current+2*np.pi
+        while theta_current>np.pi:
+            theta_current = theta_current - 2*np.pi
+
+        v_current = np.sqrt(self.state.twist.linear.x**2+self.state.twist.linear.y**2)
+
+        #self.get_logger().info("SPEED: "+str(v_current))
+
+        dist = np.zeros((1,len(self.ref_traj[:,1])))
+        for i in range(len(self.ref_traj[:,1])):
+            dist[0][i] = dist[0][i] = (x_current+np.cos(theta_current)*self.lookahead-self.ref_traj[i][0])**2+(y_current+np.sin(theta_current)*self.lookahead-self.ref_traj[i][1])**2
```

Comment:
this code is a little tough to read. if it's required, comments would go a long way.

Reply:
dist[0][i] = dist[0][i] = typo

Reply:
There is a much cleaner way of getting distances than a for loop with such a long formula. 
dist = np.linalg.norm(wpts - point, axis=1) where point is the transformed point (with the lookahead distance and transformation with the yaw) and wpts is the x and y coordinate of the array of all wpts